ip: Improve pre-edit and pre-verify code

cathay4t opened the pull request:
With recent improvement of `pre_edit_cleanup` and `pre_verify_cleanup`, we
can modify the `InterfaceIp` base on current state(or pre-applied current
state).  So there is no need to invoke these two functions at
`Interfaces` level:
 * `merge_ip_stack`
 * `include_current_ip_address_if_dhcp_on_to_off`

Both of them not is handled by `pre_edit_cleanup()` and
`pre_verify_cleanup()`.

No integration test case included as these test case will confirm the old
behaviour is unchanged:
 * `test_ipv4_dhcp_switch_on_to_off`
 * `test_ipv6_dhcp_switch_on_to_off`
 * `test_merge_ip_enabled_property_from_current`

cathay4t commented:
Wait PR #2028

ffmancera reviewed (changes requested):
Small nit on commit message: `Both of them not is handled by `pre_edit_cleanup()` and
`pre_verify_cleanup()`.` it should be `Both of them now are handled by..`. Thanks!

cathay4t committed:
ip: Improve pre-edit and pre-verify code

With recent improvement of `pre_edit_cleanup` and `pre_verify_cleanup`,
we can modify the `InterfaceIp` base on current state(or pre-applied
current state).  So there is no need to invoke these two functions at
`Interfaces` level:
 * `merge_ip_stack`
 * `include_current_ip_address_if_dhcp_on_to_off`

Both of them are can be handled by `pre_edit_cleanup()` and
`pre_verify_cleanup()`.

No integration test case included as these test case will confirm the
old behaviour is unchanged:
 * `test_ipv4_dhcp_switch_on_to_off`
 * `test_ipv6_dhcp_switch_on_to_off`
 * `test_merge_ip_enabled_property_from_current`

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
> Small nit on commit message: `Both of them not is handled by `pre_edit_cleanup()`and`pre_verify_cleanup()`.` it should be `Both of them now are handled by..`. Thanks!

Fixed. Thanks!
